scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -304,24 +304,34 @@
-                                                   ; Then the watch update occurs and we see (:missing,:starting), log a weird state,
-                                                   ; and kill it off. We want to avoid that path by doing an opportunistic kill here.
-                                                   ; That way we if we're seeing a stale :missing, we'll kill it anyways.
-                                                   ; We can't do that now because pod is nil, and kill-pod requires non-nil pod.
-
-                                                   (handle-pod-killed compute-cluster pod-name))
+                                        :missing
+                                        ; There's a race where we can launch then kill, when the kill arrives after
+                                        ; the launch, but before the watch updates k8s-actual-state.
+                                        ; (k8s-actual-state isn't the actual state, because of this watch lag)
+                                        ; When that happens, we will have an actual state of :missing.
+                                        ; If we did nothing, we'd log this as a weird state, then when the watch
+                                        ; shows up, we'd see (:missing, :starting) and log that as a weird state too.
+                                        ;
+                                        ; So, a better approach. If we detect a :missing,:killed, then we opportunistically
+                                        ; try to kill the pod. This is why update-cook-expected-state saves :launch-pod,
+                                        ; so its available here.
+                                        (if-let [pod (some-> cook-expected-state-dict :launch-pod :pod)]
+                                          (do (log/info "In compute cluster" name ", opportunistically killing" pod-name "because of potential race where kill arrives before the watch responds to the launch")
+                                              (kill-pod api-client :ignored pod))
+                                          (do
+                                            ; We treat a deleting pod in kubernetes the same as a missing pod when coming up with a synthesized state.
+                                            ; That's good for (almost) all parts of the system. However,
+                                            ; If it is legitimately missing, then something weird is going on. If it is
+                                            ; deleting, that's an expected state. So, lets be selective with our logging.
+                                            (if (= (:state synthesized-state) :missing)
+                                              (log/info "In compute cluster" name ", pod" pod-name
+                                                        "was killed with cook expected state"
```

Comment:
I don't think we need this log. We already have the same thing logged at the same level at the top of the `process` function.

Reply:
This log is here to distinguish this branch from the other branch. Each of the three subcases in this state should be distinguished clearly.

Reply:
OK sounds good